site/src/app/v2/test-eligibilite/components/radioButtonsGroup/RadioButtonsGroup.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import styles from './styles.module.scss';
+
+type Options = {
+  label: string;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}[];
+
+interface Props {
+  fieldsetId: string;
+  options: Options;
+}
+
+const RadioButtonsGroup: React.FC<Props> = ({ options, fieldsetId }) => {
+  const [buttonClickedIndex, setButtonClickedIndex] = useState<number | null>(null);
+
+  const buttonHandler = (index: number) => {
+    setButtonClickedIndex(index);
+  };
+
+  const rewireOnClick = () => {
+    return options.map((option, index) => ({
+      ...option,
+
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+        buttonHandler(index);
+        option.onChange(e);
+      },
+    })) as Options;
+  };
+
+  return (
+    <fieldset className="fr-fieldset" id={fieldsetId} aria-labelledby={`${fieldsetId}-legend `}>
+      <legend
+        className={`fr-fieldset__legend--regular fr-fieldset__legend fr-pt-1w fr-pb-2w ${styles.legend}`}
+        id={`${fieldsetId}-legend`}
+      >
+        Choisissez une option
+      </legend>
+
+      {rewireOnClick().map((option, index) => {
+        return (
+          <div key={option.label} className={`fr-fieldset__element ${styles['fieldset__element']}`}>
+            <div
+              className={buttonClickedIndex !== index ? styles['transparent-border'] : undefined}
+            >
+              <div
+                className={`fr-radio-group ${styles['radio-group']} ${buttonClickedIndex === index ? styles['selected-radio'] : styles['deselected-radio']}`}
```

Comment:
C'est illisible, je ferais ce traitement en amont de la manière suivante:


```suggestion
  const radioClass = buttonClickedIndex === index ? styles['selected-radio'] : styles['deselected-radio'];
  const radioGroupClass = `fr-radio-group ${styles['radio-group']} ${radioClass}`;


// Et ensuite
className={radioGroupClass}
```

Reply:
> C'est illisible, je ferais ce traitement en amont de la manière suivante:

je ne suis pas hyper convaincu mais je suis d'accord que la lisibilité est améliorable. Que penses tu d'utiliser ca ?https://www.npmjs.com/package/classnames  On gagnerait en lisibilité

Ca donnerait un truc comme c

```
import cn from classnames; 

.....
classNames={ 
cn( 'fr-radio-group', styles['radio-group'], 
{
  styles['selected-radio']: buttonClickedIndex === index,
  styles['deselected-radio']: buttonClickedIndex !== index,
} 
) 
}
.....
```

Reply:
Yes ca me va, j'ai déjà utilisé cette lib et elle est cool 👍

Reply:
vu ensemble, j'ai installé `classnames`